src/core/cgroup.c

```diff
@@ -817,9 +818,9 @@ int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
         if (!patterns)
                 return log_oom_debug();
 
-        memcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
-        patterns[allowed_patterns_len] = '\xff';
-        memcpy_safe(&patterns[allowed_patterns_len + 1], denied_patterns, denied_patterns_len);
+        last = mempcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
+        *last++ = '\xff';
+        mempcpy_safe(last, denied_patterns, denied_patterns_len);
```

Comment:
here you can just use plain memcpy_safe(), after all the return value doesn't matter.